src/main/kotlin/com/sourcegraph/cody/agent/protocol/ProtocolTextDocument.kt

```diff
@@ -15,32 +14,36 @@ private constructor(
 
   companion object {
 
-    private fun getSelection(editor: Editor): Range {
+    private fun getSelection(editor: Editor): Range? {
       val selectionModel = editor.selectionModel
       val selectionStartPosition =
           selectionModel.selectionStartPosition?.let { editor.visualToLogicalPosition(it) }
       val selectionEndPosition =
           selectionModel.selectionEndPosition?.let { editor.visualToLogicalPosition(it) }
-      if (selectionStartPosition != null && selectionEndPosition != null) {
+      if (selectionStartPosition != null &&
+          selectionEndPosition != null &&
+          selectionStartPosition != selectionEndPosition) {
         return Range(
             Position(selectionStartPosition.line, selectionStartPosition.column),
             Position(selectionEndPosition.line, selectionEndPosition.column))
       }
-      val caret = editor.caretModel.primaryCaret
-      val position = Position(caret.logicalPosition.line, caret.logicalPosition.column)
-      // A single-offset caret is a selection where end == start.
-      return Range(position, position)
+      return null
```

Comment:
That is actually pretty big change.
I noticed that passing empty range to simulate caret is on the agent side interpreted as a empty selection.
And e.g. in case of empty selection on empty lines, it seems to reliably cause skipping given file from the context at all (e.g. when using chat to ask the question about the current file).
So that change should improve context understanding for all our actions.

Reply:
I reverted it as it was causing problems with chat context responses, but I think this change address most corner cases:
https://github.com/sourcegraph/cody/pull/3833